apiron/__init__.py

```diff
@@ -0,0 +1,4 @@
+from apiron.endpoint import *
```

Comment:
Why `import *` instead of documenting the imports explicitly as below for `client` and `exceptions`?

Reply:
`client` and `exceptions` are modules, whereas `endpoint` and `service` are packages that define an `__all__`. The whole point of `__all__` is to enable this, really, so I figured I'd use it. I'd prefer to direct people to the actual documentation for understanding the full breadth of available features. Thoughts?

Reply:
The whole point of `__all__` is that the linter won't freak out about unused imports in my init files. 😸 

```python
from foo import bar
__all__ = ['bar']
```

Jokes aside, avoiding `*` is largely a preference on my part---no strong reasons here. I often look at inits to see what's supposed to be exposed by a module or package. It's nice when it's all in one place. Plus, someone might want to expose different things at different layers, so it makes sense that sometimes `*` won't do, while being explicit always works. That, and cruel linters finally beat a revulsion of `import *` into me.

Anyway, @daneah your rationale makes sense. I was just curious. 😄

Reply:
That's fair, and at some point in the future if it's clear we need to deliberately layer what we expose I'm all about designing that scheme. I don't prefer to use `import *` in higher-level code, but when it comes to lower-level libraries I think it serves a valid purpose.